meta-mender-core/recipes-mender/mender-client/mender-client_git.inc

```diff
@@ -54,47 +54,20 @@ SRC_URI = "git://github.com/mendersoftware/mender;protocol=https;branch=${MENDER
 # DO NOT change the checksum here without make sure that ALL licenses (including
 # dependencies) are included in the LICENSE variable below.
 def mender_license(branch):
-    if branch == "1.7.x":
-        return {
-                   "md5": "5632b9f17043c6f5f532501778595c78",
-                   "license": "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8",
-        }
-    elif branch.startswith("1."):
-        return {
-                   "md5": "debbe5e440f2e65465e86b25fc7c9fcc",
-                   "license": "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8",
-        }
-    elif branch == "2.0.x":
-        return {
-                   "md5": "ccb00e21c31df7189b2bd237ed86e7c2",
-                   "license": "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8",
-        }
-    elif branch == "2.1.x":
-        return {
-                   "md5": "1c872e33af7af0e7ee99e17f38dbcffc",
-                   "license": "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8",
-        }
-    elif branch == "2.2.x":
+    import re
+    if branch.startswith("1."):
         return {
-                   "md5": "80ba3790b689991e47685da401fd3375",
                    "license": "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8",
         }
-    elif branch == "2.3.x":
+    elif re.match(r"2\.[0-3]\.x", branch) is not None:
         return {
-                   "md5": "3e7426c258f60f9876ae3746f54c49d4",
                    "license": "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8",
         }
-    elif branch == "2.4.x":
-        return {
-                   "md5": "5649992d13a6fda40abfac7730a62b07",
-                   "license": "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8 & OpenSSL",
-        }
     else:
         return {
-                   "md5": "76d17470649a8fdd1a679a9c56a35c15",
                    "license": "Apache-2.0 & BSD-2-Clause & BSD-3-Clause & ISC & MIT & OLDAP-2.8 & OpenSSL",
         }
-LIC_FILES_CHKSUM = "file://src/github.com/mendersoftware/mender/LIC_FILES_CHKSUM.sha256;md5=${@mender_license(d.getVar('MENDER_BRANCH'))['md5']}"
+LIC_FILES_CHKSUM = "file://src/github.com/mendersoftware/mender/LICENSE;md5=7fd64609fe1bce47db0e8f6e3cc6a11d"
```

Comment:
This will need an update once a year. Quite an improvement :tada:

Reply:
Indeed!